lib/Feature/FeatureModelBuilder.cpp

```diff
@@ -0,0 +1,89 @@
+#include "vara/Feature/FeatureModelBuilder.h"
+
+namespace vara::feature {
+
+//===----------------------------------------------------------------------===//
+//                        FeatureModelBuilder
+//===----------------------------------------------------------------------===//
+
+/// Decide whether two features are mutual exclusive. Beware that this method
+/// only detects very simple trees with binary excludes.
+bool isSimpleMutex(const Feature *A, const Feature *B) {
```

Comment:
I think, we should simplify this implementation a bit by refactoring out some parts. This is quite hard to understand.